ord_schema/proto/reaction.proto

```diff
@@ -861,26 +852,104 @@ message ReactionProduct {
+    IDENTITY = 2;
+    // Yields should only be reported as quantitative yields, e.g., measured
+    // using an isolated weight analysis, quantitative NMR with an internal
+    // standard, LC peak area with an internal standard and calibration curve
+    // obtained using an authentic standard. 
+    YIELD = 3;
+    // Selectivites include enantiomeric excess/ratios, diastereomic ratios,
+    // etc. The detailed type should be specified in selectivity_type.
+    SELECTIVITY = 4;
+    // The purity of the product mixture, normalized to all other species
+    // observed in the analysis.
+    PURITY = 5;
+    // Integrated peak area, e.g., as in an LC chromatogram.
+    AREA = 6;
+    // Retention time, e.g., as in an LC or GC chromatogram.
+    RETENTION_TIME = 7;
+    // Raw counts, e.g., as in an EIC or TIC from a MS analysis. If an EIC,
+    // the m/z ratio(s) should be defined in the details.
+    COUNTS = 8;
+    // Intensity, e.g., of a TLC spot, ELSD, or GC FID.
+    INTENSITY = 9;
+    // Direct m/z measurement, e.g., from HRMS. The value field should be used
+    // to report the m/z, not the "mass" field.
+    MZ = 10;
+    // Temperature, e.g., from a melting point experiment.
+    TEMPERATURE = 11;
+    // Angle, e.g., from a circular dichroism experiment. Units of degrees
+    // centimer squared per decimol are assumed.
+    ANGLE = 12;
   }
-  SelectivityType type = 1;
-  string details = 2;
-  optional float value = 3;
-  // TODO(kearnes): What does precision mean in this context?
-  optional float precision = 4;
+  MeasurementType type = 2;
+  // Additional information about the measurement and, in particular, how it
+  // was calculated if that is ambiguous given the analysis type.
+  string details = 3;
+  // Whether an internal standard was used for this measurement to normalize
+  // it, e.g., for quantification of yield or relative peak area.
+  optional bool uses_internal_standard = 4;
+  // Whether an authentic sample of a material was used for callibrating
+  // an analytical technique, e.g., for product identification by LCMS.
+  // "Authentic" refers to a sample of a compound whose structure has been
+  // proven by the accepted means (1H NMR, 13C NMR, IR, HRMS), and is now
+  // being used as a comparison for other analytical methods.
+  optional bool uses_authentic_standard = 5;
+  // Details about the authentic standard used, optionally including
+  // information about its precise salt form and source/vendor.
+  Compound authentic_standard = 6;
+  // The numeric value for this measurement. For example, the yield expressed
+  // as a percentage (i.e., 50% yield corresponds to 50 and not 0.5). If the
+  // value is associated with units, those unit-specific fields should be used
+  // instead.
+  optional float value = 7;
+  optional float precision = 8;
+  // The value for this measurement metric when it cannot be represented by a
+  // single number.
+  string string_value = 9;
+  Mass mass = 10;
```

Comment:
Is this meant for a mass obtained if the Measurement is `YIELD` or should this only be included in `Compound amount`? Or, is it mass found referring to `MZ` in HRMS? Both the found and calculated masses are usually reported for HRMS, are we going to try to handle this or just only report the found? Suppose this could be a reason for structured messages for the analysis types.

Reply:
That mass was meant to be for isolated yield, not m/z. 

That's a good question about HRMS. I would consider the calculated mass a feature of the compound itself, while the found is associated with a specific analysis/product/measurement. However, defining those values in two separate messages would be rather confusing and unintuitive. 

This does make it more attractive to define a complete set of structured messages for common analyses...

Reply:
No longer relevant since we've removed structured analytical data from the ProductMeasurement

Reply:
After discussion, decided to avoid structured analytical data messages for now and stick with the unstructured Data fields in the analysis (with no data other than derived product-specific statistics in ProductMeasurement)